rfcs/0124-worker-manager.md

```diff
@@ -0,0 +1,271 @@
+# RFC 124 - Worker Manager
+* Comments: [#124](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/124)
+* Proposed by: @jhford
+
+# Summary
+Taskcluster was originally designed with a single Queue instance in the world.
+This single Queue would be used to coordinate all tasks from all Taskcluster
+users.  Isolation between different users of Taskcluster would be provided by
+using different provisioners and workers.
+
+Over time, the deployment model of Taskcluster has changed to being a
+redeployable software system.  The idea of having one Queue and many users has
+migrated towards having one Queue for each Taskcluster environment.  For the
+Firefox-CI use case, this means possibly having a full production cluster
+alongside one or more staging clusters.
+
+There has also been difficulty in building coherent and reliable worker
+management tools which span more than one cloud.  The Queue is the single
+coordinating system of Taskcluster, but is not an expert at managing workers.
+The systems which *are* experts at worker management do not have a
+programmatically discoverable API.
+
+We would like to change the architecture of Taskcluster to have a new component
+called the worker manager.  This new component would take over the worker
+management responsibilities from the Queue as well as providing a single API to
+manage workers.
+
+The worker manager would be a single instance per cluster.  It would be
```

Comment:
Perhaps here we state that this is similar to tc-queue and tc-auth? It could be a rule we set: "platform services are one-offs"?

Actually, with the way we've implemented routing now, I think _every_ service is a one-off.

Reply:
sure!